File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        HashMap<Character,Character> hm=new HashMap<>();
        for(int i=0;i<s.length();i++){
            char chars=s.charAt(i);
            char chart=t.charAt(i);
            if(hm.containsKey(chars)){
                if(hm.get(chars)!=chart){
                    return false;
                }
            }
            else{
                if(hm.containsValue(chart)){
                    return false;
                }
                hm.put(chars,chart);
            }
            
        }
        return true; 
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False
        if(len(set(s))!=len(set(t))):
            return False
        dic={}
        for i in range(len(s)):
            if s[i] not in dic:
                dic[s[i]]=t[i]
            elif dic[s[i]]!=t[i]:
                return False
        return True